tokio/src/time/driver/sleep.rs

```diff
@@ -244,37 +295,79 @@ impl Sleep {
     ///
     /// [`Pin::as_mut`]: fn@std::pin::Pin::as_mut
     pub fn reset(self: Pin<&mut Self>, deadline: Instant) {
+        self.reset_inner(deadline)
+    }
+
+    fn reset_inner(self: Pin<&mut Self>, deadline: Instant) {
         let me = self.project();
         me.entry.reset(deadline);
-        *me.deadline = deadline;
+        (*me.inner).deadline = deadline;
+
+        #[cfg(all(tokio_unstable, feature = "tracing"))]
+        {
+            let now = me.inner.time_source.now();
+            let duration = me.inner.time_source.deadline_to_tick(deadline) - now;
+            let _span_guard = me.inner.resource_span.enter();
+            tracing::trace!(
+                target: "runtime::resource::state_update",
+                duration = duration,
+                duration.unit = "ms",
+                duration.op = "override",
```

Comment:
nit/TIOLI: it would be nice to avoid the `Instant::now` call if the `tracing` event is disabled by tracing filtering. The resulting code is admittedly a bit uglier, but if we stuck the duration calculation inside the `tracing` macro, it would only be evaluated if the event is enabled:
```suggestion
            let _span_guard = me.inner.resource_span.enter();
            tracing::trace!(
                target: "runtime::resource::state_update",
                duration = {
                    let now = me.inner.time_source.now();
                    me.inner.time_source.deadline_to_tick(deadline) - now
                },
                duration.unit = "ms",
                duration.op = "override",
```